apps/matrix/main.py

```diff
@@ -148,16 +154,26 @@ def display_rooms(self):
 			# Right arrow 	-> Write message to room
 			menu_contents.append([
 				room_name,
-				lambda x=current_room: self.display_messages(x),
-				lambda x=current_room: self.write_message(x)
+				lambda r=current_room: self.display_messages(r),
+				lambda r=current_room: self.display_room_members(r)
 			])
 
 		menu_contents.append(["Settings", self.show_settings])
 		Menu(menu_contents, self.i, self.o, name="Matrix app main menu").activate()
 
+	# Display a menu of settings for the app
 	def show_settings(self):
-		mc = [["Log out", self.logout]]
-		Menu(mc, self.i, self.o, catch_exit=False, name="Matrix app settings menu").activate()
+
+		def gen_menu_contents():
+			mc = [["Log out", self.logout]]
+			mc.append(["Hide join/leave messages", lambda: self._change_setting("show_join_leave_messages", False)] if self.config["show_join_leave_messages"] else ["Show join/leave messages", lambda: self._change_setting("show_join_leave_messages", True)])
```

Comment:
The line is too long, we can shorten it like this:
```suggestion
			conf_key = "show_join_leave_messages"
			mc.append([ "Hide join/leave messages", \
						lambda: self._change_setting(conf_key, False) ] \
						if self.config[conf_key] else [ "Show join/leave messages", \
						lambda: self._change_setting(conf_key, True) ])
```
Do test that it works after merging if you can, please =)